### ISSUE TITLE
brace-style does not validate the opening curly brace in case statement

### ISSUE THREAD

**Issue by buryndin** [eslint/eslint#9749]
* **ESLint Version:**
4.13.1
* **Node Version:**
v6.11.0

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**
```js
/* eslint brace-style: error */

switch (foo) {
  case 1: 
  {
    foo = 0;
  }
}

```
**What did you expect to happen?**

I expected one error to be reported for the brace { after case.

**What actually happened? Please include the actual, raw output from ESLint.**

No errors were reported.

**Comment by not-an-aardvark** [comment]
Thanks for the report; however, this is working as intended. The difference between syntax like `if`, `while`, and `for` versus syntax like `case` is that an `if` statement always has a single statement as the body (which is often a block statement composed of multiple smaller statements). On the other hand, `switch` cases can contain multiple statements, so the `brace-style` rule doesn't apply to them in general.

To illustrate the point, the following code would also be valid syntax:

```js
switch (foo) {
  case 1:
    doSomething();
    {
      foo = 0;
    }
    doSomethingElse();
}
```

In the above example, there isn't really a good way to format the block that would be consistent with what the `brace-style` rule normally enforces.

I suppose we could add a special case to the `brace-style` rule so that it enforces the style when there is exactly one statement in a switch case. But I'm unsure if that would be confusing to users.

**Comment by not-an-aardvark** [comment]
Closing because this is working as intended.
